src/main/java/teammates/common/util/MainAppUrl.java

```diff
@@ -0,0 +1,13 @@
+package teammates.common.util;
+
+/**
+ * An implementation of {@link AppUrl}, with the base URL pre-set as 
+ * the value of app.url in build.properties.
+ */
+public class MainAppUrl extends AppUrl {
+
+    public MainAppUrl(String url) {
+        super(Config.APP_URL, url);
+    }
+
+}
```

Comment:
I can't shake the feeling that `MainAppUrl` should be an object not a class. Hence my push for factory methods. Having factory methods instead of these classes reduces class count by 2 and reverses the `*Url` --> `Config` dependency both of which are good things.